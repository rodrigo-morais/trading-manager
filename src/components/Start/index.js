import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import TextBox from '../shared/TextBox'


const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 24% 24% 24% 24%;
  justify-content: space-evenly;
  min-width: 100%;
`

const Button = styled.button`
  width: 50%;
`

const Start = ({ className, disabled }) => (
  <Container className={className}>
    <TextBox label="Entrada" disabled={disabled} />
    <TextBox label="Stop Base" disabled={disabled} />
    <TextBox label="Hora" disabled={disabled} />
    <Button value="Iniciar" disabled={disabled}>Iniciar</Button>
  </Container>
)

Start.defaultProps = {
  className: '',
  disabled: false,
}

Start.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
}

export default Start
