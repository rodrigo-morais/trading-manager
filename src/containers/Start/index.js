import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import TextBox from '../../components/shared/TextBox'

const Container = styled.div`
  display: inline-grid;
  grid-template-columns: 24% 24% 24% 24%;
  justify-content: space-evenly;
  min-width: 100%;
`

const Button = styled.button`
  width: 50%;
  font-size: 14px;
  font-weight: bold;
`

class Start extends React.Component {
  constructor(props) {
    super(props)

    this.changeStartValue = this.changeStartValue.bind(this)
    this.changeInitialStopLoss = this.changeInitialStopLoss.bind(this)
    this.changeTime = this.changeTime.bind(this)
    this.onClick = this.onClick.bind(this)

    this.state = {
      startValue: '',
      initialStopLoss: '',
      time: '',
    }
  }

  onClick(callback, event) {
    event.preventDefault()
    const { startValue, initialStopLoss, time } = this.state

    callback(startValue, initialStopLoss, time)
  }

  changeStartValue(value) {
    this.setState(() => ({ startValue: value }))
  }

  changeInitialStopLoss(value) {
    this.setState(() => ({ initialStopLoss: value }))
  }

  changeTime(value) {
    this.setState(() => ({ time: value }))
  }

  render() {
    const { trades, className, disabled, startTrading } = this.props

    return (
      <Container className={className}>
        <TextBox label="Entrada" disabled={disabled} value={trades.startValue} onChange={this.changeStartValue} />
        <TextBox label="Stop Base" disabled={disabled} value={trades.initialStopLoss} onChange={this.changeInitialStopLoss} />
        <TextBox label="Hora" disabled={disabled} value={trades.time} onChange={this.changeTime} />
        <Button value="Iniciar" disabled={disabled} onClick={(event) => this.onClick(startTrading, event)}>Iniciar</Button>
      </Container>
    )
  }
}

Start.defaultProps = {
  className: '',
  disabled: false,
}

Start.propTypes = {
  trades: PropTypes.object.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  startTrading: PropTypes.func.isRequired,
}

export default Start
