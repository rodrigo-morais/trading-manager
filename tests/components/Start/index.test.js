import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Start from '../../../src/components/Start'

describe('Start', () => {
  describe('Rendering', () => {
    const wrapper = (
      <Start />
    )

    test('renders according to design', () => {
      const component = renderer.create(wrapper)
      const tree = component.toJSON()

      expect(tree).toMatchSnapshot()
    })

    test('first label as "Entrada"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(0).text()).toEqual('Entrada:')
    })

    test('second label as "Stop Base"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(1).text()).toEqual('Stop Base:')
    })

    test('third label as "Hora"', () => {
      const component = mount(wrapper)
      expect(component.find('label').at(2).text()).toEqual('Hora:')
    })

    test('button', () => {
      const component = mount(wrapper)
      expect(component.find('button').length).toEqual(1)
      expect(component.find('button').at(0).text()).toEqual('Iniciar')
    })
  })
})
