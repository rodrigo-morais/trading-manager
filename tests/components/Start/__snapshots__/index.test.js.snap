// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Rendering renders according to design 1`] = `
<div
  className="sc-bxivhb dvxyIR"
>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Entrada"
    >
      Entrada:
    </label>
    <input
      className="sc-htpNat iZIyaN"
      disabled={false}
      id="Entrada"
      name="Entrada"
      type="text"
    />
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Stop Base"
    >
      Stop Base:
    </label>
    <input
      className="sc-htpNat iZIyaN"
      disabled={false}
      id="Stop Base"
      name="Stop Base"
      type="text"
    />
  </div>
  <div
    className="sc-bdVaJa lhrPED"
  >
    <label
      className="sc-bwzfXH ebnOHd"
      htmlFor="Hora"
    >
      Hora:
    </label>
    <input
      className="sc-htpNat iZIyaN"
      disabled={false}
      id="Hora"
      name="Hora"
      type="text"
    />
  </div>
  <button
    className="sc-ifAKCX loRasB"
    disabled={false}
    value="Iniciar"
  >
    Iniciar
  </button>
</div>
`;
